query/src/test/java/org/infinispan/query/projection/ScoreProjectionTest.java

```diff
@@ -0,0 +1,104 @@
+import org.infinispan.commons.api.query.Query;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.query.model.Game;
+import org.infinispan.search.mapper.mapping.SearchMapping;
+import org.infinispan.search.mapper.scope.SearchScope;
+import org.infinispan.search.mapper.session.SearchSession;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "query.projection.ScoreProjectionTest")
+public class ScoreProjectionTest extends SingleCacheManagerTest {
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      ConfigurationBuilder indexed = new ConfigurationBuilder();
+      indexed.indexing().enable()
+            .storage(LOCAL_HEAP)
+            .addIndexedEntity(Game.class);
+
+      return TestCacheManagerFactory.createCacheManager(indexed);
+   }
+
+   @BeforeMethod
+   public void setUp() {
+      if (!cache.isEmpty()) {
+         return;
+      }
+
+      cache.put(4, new Game("4", "descriAAion"));
+      cache.put(1, new Game("1", "description"));
+      cache.put(3, new Game("3", "desription"));
+      cache.put(5, new Game("5", "very-different"));
+      cache.put(2, new Game("2", "descriqtion"));
+   }
+
+   @Test
+   public void entityAndScore() {
+      Query<Object[]> query;
+      List<Object[]> games;
+
+      query = cache.query("select g, score(g) from org.infinispan.query.model.Game g where g.description : 'description'~2");
+      games = query.list();
+      assertThat(games).extracting(objects -> objects[0]).extracting("name").containsExactly("1", "2", "3", "4");
+      assertThat(games).extracting(objects -> objects[1]).hasOnlyElementsOfType(Float.class).isNotNull();
+   }
+
+   @Test
+   public void fieldAndScore() {
+      Query<Object[]> query;
+      List<Object[]> games;
+
+      query = cache.query("select score(g), g.name from org.infinispan.query.model.Game g where g.description : 'description'~2");
+      games = query.list();
+      assertThat(games).extracting(objects -> objects[1]).containsExactly("1", "2", "3", "4");
+      assertThat(games).extracting(objects -> objects[0]).hasOnlyElementsOfType(Float.class).isNotNull();
+   }
+
+   @Test
+   public void entityScoreAndField() {
+      Query<Object[]> query;
+      List<Object[]> games;
+
+      query = cache.query("select g.name, score(g), g from org.infinispan.query.model.Game g where g.description : 'description'~2");
+      games = query.list();
+      assertThat(games).extracting(objects -> objects[0]).containsExactly("1", "2", "3", "4");
+      assertThat(games).extracting(objects -> objects[1]).hasOnlyElementsOfType(Float.class).isNotNull();
```

Comment:
in langchain, the score is a double

Reply:
only 4 test pass now

```[ERROR] dev.langchain4j.store.embedding.infinispan.InfinispanEmbeddingStoreIT.should_return_correct_score -- Time elapsed: 0.430 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual:
  0.7332834601402283
to be close to:
  0.9090677379700546
by less than 1% but difference was 19.336763421211277%.
(a difference of exactly 1% being considered valid)
	at dev.langchain4j.store.embedding.EmbeddingStoreWithoutMetadataIT.should_return_correct_score(EmbeddingStoreWithoutMetadataIT.java:237)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

[ERROR] dev.langchain4j.store.embedding.infinispan.InfinispanEmbeddingStoreIT.should_find_with_min_score -- Time elapsed: 0.155 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual:
  0.7332834601402283
to be close to:
  0.9090677379700546
by less than 1% but difference was 19.336763421211277%.
(a difference of exactly 1% being considered valid)
	at dev.langchain4j.store.embedding.EmbeddingStoreWithoutMetadataIT.should_find_with_min_score(EmbeddingStoreWithoutMetadataIT.java:188)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

[ERROR] dev.langchain4j.store.embedding.infinispan.InfinispanEmbeddingStoreIT.should_add_multiple_embeddings_with_segments -- Time elapsed: 0.152 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual:
  0.7332834601402283
to be close to:
  0.9090677379700546
by less than 1% but difference was 19.336763421211277%.
(a difference of exactly 1% being considered valid)
	at dev.langchain4j.store.embedding.EmbeddingStoreWithoutMetadataIT.should_add_multiple_embeddings_with_segments(EmbeddingStoreWithoutMetadataIT.java:160)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

[ERROR] dev.langchain4j.store.embedding.infinispan.InfinispanEmbeddingStoreIT.should_add_multiple_embeddings -- Time elapsed: 0.145 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual:
  0.7332834601402283
to be close to:
  0.9090677379700546
by less than 1% but difference was 19.336763421211277%.
(a difference of exactly 1% being considered valid)
	at dev.langchain4j.store.embedding.EmbeddingStoreWithoutMetadataIT.should_add_multiple_embeddings(EmbeddingStoreWithoutMetadataIT.java:121)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   InfinispanEmbeddingStoreIT>EmbeddingStoreWithoutMetadataIT.should_add_multiple_embeddings:121 
Expecting actual:
  0.7332834601402283
to be close to:
  0.9090677379700546
by less than 1% but difference was 19.336763421211277%.
(a difference of exactly 1% being considered valid)
[ERROR]   InfinispanEmbeddingStoreIT>EmbeddingStoreWithoutMetadataIT.should_add_multiple_embeddings_with_segments:160 
Expecting actual:
  0.7332834601402283
to be close to:
  0.9090677379700546
by less than 1% but difference was 19.336763421211277%.
(a difference of exactly 1% being considered valid)
[ERROR]   InfinispanEmbeddingStoreIT>EmbeddingStoreWithoutMetadataIT.should_find_with_min_score:188 
Expecting actual:
  0.7332834601402283
to be close to:
  0.9090677379700546
by less than 1% but difference was 19.336763421211277%.
(a difference of exactly 1% being considered valid)
[ERROR]   InfinispanEmbeddingStoreIT>EmbeddingStoreWithoutMetadataIT.should_return_correct_score:237 
Expecting actual:
  0.7332834601402283
to be close to:
  0.9090677379700546
by less than 1% but difference was 19.336763421211277%.
(a difference of exactly 1% being considered valid)
[ERROR] Errors: 
[ERROR]   InfinispanEmbeddingStoreIT>EmbeddingStoreWithoutMetadataIT.beforeEach:24->clearStore:45 » Transport io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/[0:0:0:0:0:0:0:1]:11222
[INFO] 
[ERROR] Tests run: 8, Failures: 4, Errors: 1, Skipped: 0
[INFO] 
```

Reply:
Hibernate Search returns a float, as Lucene does. I don't know if it makes sense to cast it to double from our side...

Reply:
I'm guessing Katia can do that in the Langchain4j SPI implementation?

Anyway, the problem here seems to be deeper: the test expects a certain value for the score, and that's not what it gets. No idea why... ? How was that expected score value determined?

Reply:
@yrodiere IIUC we only need to change the distance function to cosine and after that all tests pass